clients/vscode/src/chat/BaseChatView.ts

```diff
@@ -0,0 +1,342 @@
+import { ExtensionContext, WebviewView, workspace, Uri, LogOutputChannel, TextEditor, WebviewPanel } from "vscode";
+import type { ServerApi, ChatMessage, Context } from "tabby-chat-panel";
+import hashObject from "object-hash";
+import * as semver from "semver";
+import type { ServerInfo } from "tabby-agent";
+import type { AgentFeature as Agent } from "../lsp/AgentFeature";
+import { GitProvider } from "../git/GitProvider";
+
+export abstract class BaseChatView {
+  webview?: WebviewView | WebviewPanel;
+  client?: ServerApi;
+  protected pendingMessages: ChatMessage[] = [];
+  protected pendingRelevantContexts: Context[] = [];
+  protected isChatPageDisplayed = false;
+
```

Comment:
function initializeWebview(...) {
   this.webview = ...;
   this.webview.options = {...}
}